### PR TITLE
Added SQL script to spProduct_GetById and modified spProducts_Get

### PR DESCRIPTION
**Pull request by sviastyn**
Added API ProductsGet and API ProductsAttributesValues

Signed-off-by: sviastyn <svyastynvasul@gmail.com>

### CODE REVIEW

File: src/TechMarket.API/TechMarket.API/Controllers/ProductsController.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        public IActionResult Get([FromQuery] int skip = 0, [FromQuery] int take = 25)
+        {
+            IEnumerable<Product> products = this._productsService.Get(skip, take);
+            return base.Ok(new GetProductsResponse { Products = products });
+        }
+
+        [HttpGet("{Id:int}")]
+        public IActionResult GetById([FromQuery] int id)
+        {
+            //todo Validation
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Controllers/ProductsController.cs:32]: You have added simple validation, so, you don't need this comment.

File: src/TechMarket.API/TechMarket.API/Controllers/ProductsController.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            if (id <= 0)
+                return base.BadRequest(new ProblemDetails { Detail = "Not valid product id." });
+
+            var product = this._productsService.GetById(id);
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Controllers/ProductsController.cs:36]: Please, specify the explicit type and then check the result on null.

File: src/TechMarket.API/TechMarket.API/Controllers/ProductsController.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+        [HttpGet]
+        public IActionResult Get([FromQuery] int skip = 0, [FromQuery] int take = 25)
+        {
+            IEnumerable<Product> products = this._productsService.Get(skip, take);
+            return base.Ok(new GetProductsResponse { Products = products });
+        }
+
+        [HttpGet("{Id:int}")]
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Controllers/ProductsController.cs:29]: Typo - id

File: src/TechMarket.API/TechMarket.API/Controllers/ProductsController.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        public IActionResult GetById([FromQuery] int id)
+        {
+            //todo Validation
+            if (id <= 0)
+                return base.BadRequest(new ProblemDetails { Detail = "Not valid product id." });
+
+            var product = this._productsService.GetById(id);
+
+            return base.Ok(product);
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Controllers/ProductsController.cs:38]: Please, create a response type for this.

File: src/TechMarket.API/TechMarket.API/Infrastructure/Domain/ProductAttributeValue.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿namespace TechMarket.API.Infrastructure.Domain
+{
+    public class ProductAttributeValue
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public int ValueId { get; set; }
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Infrastructure/Domain/ProductAttributeValue.cs:7]: Consider having ProductAttribute name

File: src/TechMarket.API/TechMarket.API/Infrastructure/SQL/Programmability/Stored Procedures/Products/spProduct_GetById.sql [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        DROP PROCEDURE dbo.spProduct_GetById;
+    END
+GO
+------------------------------------------------------------------------------
+SET ANSI_NULLS ON;
+SET QUOTED_IDENTIFIER ON;
+SET ANSI_PADDING ON;
+GO
+--============================================================================
+CREATE PROCEDURE spProduct_GetById
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Infrastructure/SQL/Programmability/Stored Procedures/Products/spProduct_GetById.sql:22]: spProduct**s**

File: src/TechMarket.API/TechMarket.API/Infrastructure/SQL/Programmability/Stored Procedures/Products/spProduct_GetById.sql [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    @Id INT
+AS
+BEGIN
+    SET NOCOUNT ON;
+
+    --========================================================================
+    -- Validation:
+    --========================================================================
+    IF @Id IS NULL OR @Id < 0
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Infrastructure/SQL/Programmability/Stored Procedures/Products/spProduct_GetById.sql:31]: <=

File: src/TechMarket.API/TechMarket.API/Infrastructure/Services/ProductsService.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+        public IEnumerable<Product> Get(int skip, int take)
+        {
+            return this._productsRepository.Get(skip, take);
+        }
+
+        public Product GetById(int id)
+        {
+            var product = this._productsRepository.GetById(id);
+            //TODO Logic
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Infrastructure/Services/ProductsService.cs:25]: Remove TODO comment.

File: src/TechMarket.API/TechMarket.API/Startup.cs [hunk 9]
```diff
@@ -36,19 +36,25 @@ public void Configure(IApplicationBuilder appBuilder, IWebHostEnvironment hostEn
                 .UseEndpoints(endpoints => endpoints.MapControllers());
 
         private IServiceCollection ConfigureCORS(IServiceCollection services)
-            => services
+        {
+            return services
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Startup.cs:40]: Please, revert your changes here.

File: src/TechMarket.API/TechMarket.API/TechMarket.API.csproj [hunk 10]
```diff
@@ -1,4 +1,4 @@
-﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+<Project Sdk="Microsoft.NET.Sdk.Web">
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/TechMarket.API.csproj:1]: Revert this change.

**Review by MorphineForOdin — CHANGES_REQUESTED** [review]

File: src/TechMarket.API/TechMarket.API/Infrastructure/SQL/Programmability/Stored Procedures/Products/spProduct_GetById.sql (hunk 6 above)
Comment (sviastyn) [reply src/TechMarket.API/TechMarket.API/Infrastructure/SQL/Programmability/Stored Procedures/Products/spProduct_GetById.sql:22]: but we take only one product by this procedure

File: src/TechMarket.API/TechMarket.API/Infrastructure/Domain/ProductAttributeValue.cs (hunk 5 above)
Comment (sviastyn) [reply src/TechMarket.API/TechMarket.API/Infrastructure/Domain/ProductAttributeValue.cs:7]: Do you mean add "ProductAttribute " at all this properties, or only at "Name"?

Comment (MorphineForOdin) [reply src/TechMarket.API/TechMarket.API/Infrastructure/Domain/ProductAttributeValue.cs]: Rename the file itself

File: src/TechMarket.API/TechMarket.API/Infrastructure/Domain/ProductAttribute.cs [hunk 11]
```diff
@@ -0,0 +1,10 @@
+﻿namespace TechMarket.API.Infrastructure.Domain
+{
+    public class ProductAttribute
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public int ValueId { get; set; }
+        public string Value { get; set; }
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Infrastructure/Domain/ProductAttribute.cs:8]: By our design, each attribute could have several values.

File: src/TechMarket.API/TechMarket.API/Infrastructure/Repositories/DTOs/ProductDto.cs [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+{
+    public class ProductDto
+    {
+        public int Id { get; set; }
+        public int CategoryId { get; set; }
+        public string Name { get; set; }
+        public string Description { get; set; }
+        public string ImageUrl { get; set; }
+        public decimal Price { get; set; }
+        public IEnumerable<ProductAttributeValueDto> Properties { get; set; }
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Infrastructure/Repositories/DTOs/ProductDto.cs:16]: You don't need this property here.

File: src/TechMarket.API/TechMarket.API/Infrastructure/Services/ProductsService.cs [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+
+        public IEnumerable<Product> Get(int skip, int take)
+        {
+            return this._productsRepository.Get(skip, take);
+        }
+
+        public Product GetById(int id)
+        {
+            var product = this._productsRepository.GetById(id);
+            return product;
```
Comment (MorphineForOdin) [inline src/TechMarket.API/TechMarket.API/Infrastructure/Services/ProductsService.cs:25]: Please, simplify this code by returning the repo result.

**Review by MorphineForOdin — CHANGES_REQUESTED** [review]

**Comment by sviastyn** [comment]
no need

**Pull request closed without merging**